lib/atum/core/errors.rb

```diff
@@ -5,14 +5,34 @@ class SchemaError < StandardError; end
     class ResponseError < StandardError; end
 
     class ApiError < StandardError
-      attr_reader :error
+      attr_reader :request, :response
 
-      def initialize(error)
+      def initialize(request: request, response: response)
+        @request = request
+        @response = response
         @error = error
-        if @error.key?('documentation_url')
-          super("#{@error['message']}, see #{@error['documentation_url']}")
+      end
+
+      def error
+        @error ||=
+          if response.json?
+            response.body['error']
```

Comment:
Wondering whether we may as well just use `response.body` here on the off chance we don't get back what we're expecting, i.e. no `error` field. I'm not sure whether that's ever going to happen since if an error gets returned not through our error serializer, its also likely not going to be json. What do you think? Might be really handy if it every happens!